packages/lesswrong/client/start.tsx

```diff
@@ -37,14 +38,15 @@ onStartup(() => {
     />
   );
 
-  ReactDOM.hydrate(
+  const root = hydrateRoot(
+    document.getElementById('react-app')!,
     <Main />,
-    document.getElementById('react-app'),
-    () => {
-      apolloClient.disableNetworkFetches = false;
-      foreignApolloClient.disableNetworkFetches = false;
-      timeOverride.currentTime = null;
-    }
   );
+  setTimeout(() => {
+    // TODO: Check: is this too early?
```

Comment:
What's the potential issue here?

Reply:
The concern was that with react async rendering, it might run this before hydration is actually finished, leading to spurious graphql requests and SSR mismatches. It wasn't a problem during testing, I just forgot to take out the comment, doing that now.

Reply:
The concern here was that with React async rendering, I wasn't sure of the timing of when `hydrateRoot` finishes (and thus the timing of when it's safe to update `apolloClient.disableNetworkFetches` without getting spurious requests, and turn off `timeOverride` without getting SSR mismatches). Empirically, this single setTimeout seems to be sufficient, though (and I've checked whether common pages have any spurious graphql requests in them), so, taking out the comment.